core/variant/method_ptrcall.h

```diff
@@ -94,12 +99,13 @@ struct PtrToArg {};
 		_FORCE_INLINE_ static m_type convert(const void *p_ptr) {             \
 			return *reinterpret_cast<const m_type *>(p_ptr);                  \
 		}                                                                     \
+		typedef m_type EncodeT;                                               \
 		_FORCE_INLINE_ static void encode(const m_type &p_val, void *p_ptr) { \
 			*((m_type *)p_ptr) = p_val;                                       \
 		}                                                                     \
 	}
 
-MAKE_PTRARG(bool);
+MAKE_PTRARGCONV(bool, uint32_t);
```

Comment:
Any reason to use `uint32_t` instead of `uint8_t`? What do most compilers tend to use for bool?
Also does the compiler optimize this when the size is the same for both? If not we should probably use `if constexpr`.

Reply:
problem with bool is that, even if its uint8, most platforms will prefer 32 bit alignment so I would rather leave it as 32.